### PR TITLE
[Feature][web] added user list for dinky-web module

### PR DESCRIPTION
**Pull request by Zzm0809**
## Purpose of the pull request

- added user list
- added theme check

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by Zzm0809** [comment]
image

image

image

image


image


image

File: dinky-admin/src/main/java/org/dinky/controller/UserController.java [hunk 1]
```diff
@@ -74,50 +80,70 @@ public Result<Void> saveOrUpdate(@RequestBody User user) {
...
-                return Result.succeed("删除成功");
-            } else {
-                return Result.succeed("删除部分成功，但" + error + "删除失败，共" + error.size() + "次失败。");
-            }
+    public Result<Void> deleteUserById(@RequestParam("id") Integer id) {
+        if (userService.removeUser(id)) {
+            return Result.succeed("删除成功");
         } else {
-            return Result.failed("请选择要删除的记录");
+            return Result.failed("删除失败");
```
Comment (aiwenmo) [inline dinky-admin/src/main/java/org/dinky/controller/UserController.java]: Use English.

File: dinky-admin/src/main/java/org/dinky/controller/UserController.java [hunk 2]
```diff
@@ -74,50 +80,70 @@ public Result<Void> saveOrUpdate(@RequestBody User user) {
...
+     *
+     * @param user {@link User}
+     * @return {@link Result} with {@link User}
+     */
     @PostMapping("/getOneById")
     public Result<User> getOneById(@RequestBody User user) {
         user = userService.getById(user.getId());
         user.setPassword(null);
         return Result.succeed(user, "获取成功");
     }
```
Comment (aiwenmo) [inline dinky-admin/src/main/java/org/dinky/controller/UserController.java]: Use English.

File: dinky-admin/src/main/java/org/dinky/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -195,30 +192,26 @@ public User getUserByUsername(String username) {
...
+        // save or update user role
+        boolean result = userRoleService.saveOrUpdateBatch(userRoleList, 1000);
+        if (result) {
+            return Result.succeed("用户授权角色成功");
         } else {
-            return Result.failed("请选择要授权的角色");
+            if (userRoleList.size() == 0) {
+                return Result.succeed("该用户绑定的角色已被全部删除");
+            }
+            return Result.failed("用户授权角色失败");
```
Comment (aiwenmo) [inline dinky-admin/src/main/java/org/dinky/service/impl/UserServiceImpl.java]: Use English.

**Comment by aiwenmo** [comment]
Why use two tone icon?

File: dinky-web/src/components/Icons/CustomIcons.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import {DeleteTwoTone} from "@ant-design/icons";
+
+export const CustomDeleteIcon = () => {
+  return <>
```
Comment (aiwenmo) [inline dinky-web/src/components/Icons/CustomIcons.tsx]: DangerDeleteIcon has a clearer meaning than CustomDeleteIcon.

**Review by aiwenmo — CHANGES_REQUESTED** [review]
Is it compatible dlink-web?

**Comment by Zzm0809** [comment]
> Is it compatible dlink-web?

If compatibility is required, then refactor directly on dlink-web, otherwise, both web modules need to be modified at the same time

File: dinky-web/src/components/Icons/CustomIcons.tsx (hunk 4 above)
Comment (Zzm0809) [reply dinky-web/src/components/Icons/CustomIcons.tsx]: > DangerDeleteIcon has a clearer meaning than CustomDeleteIcon.

ok i will be modify

**Comment by Zzm0809** [comment]
> Why use two tone icon?

I think using TwoTone icons makes the page more beautiful. If TwoTone Icons can’t satisfy you, you can add a custom Icon

**Review by aiwenmo — APPROVED** [review]
LGTM

**Comment by leechor** [comment]
LGTM

**Pull request merged**